libs/wingc/src/type_check.rs

```diff
@@ -1412,21 +1459,59 @@ impl<'a> TypeChecker<'a> {
 				}
 				self.validate_type(exp_type, var_info._type, value);
 			}
-			StmtKind::Use {
+			StmtKind::Bring {
 				module_name,
 				identifier,
 			} => {
-				_ = {
-					// If provided use alias identifier as the namespace name
-					let namespace_name = identifier.as_ref().unwrap_or(module_name);
-
-					if namespace_name.name == WINGSDK_STD_MODULE {
-						self.stmt_error(stmt, format!("Redundant import of \"{}\"", WINGSDK_STD_MODULE));
+				// library_name is the name of the library we are importing from the JSII world
+				let library_name: String;
+				// namespace_filter describes what types we are importing from the library
+				// e.g. [] means we are importing everything from `mylib`
+				// e.g. ["ns1", "ns2"] means we are importing everything from `mylib.ns1.ns2`
+				let namespace_filter: Vec<String>;
+				// alias is the symbol we are giving to the imported library or namespace
+				let alias: &Symbol;
+
+				if module_name.name.starts_with('"') && module_name.name.ends_with('"') {
+					// case 1: bring "library_name" as identifier;
+					if identifier.is_none() {
+						self.stmt_error(
+							stmt,
+							format!("bring \"{}\" must be assigned to an identifier", module_name.name),
+						);
 						return;
 					}
+					// We assume we have a jsii library and we use `module_name` as the library name, and set no
+					// namespace filter (we only support importing a full library at the moment)
+					library_name = module_name.name[1..module_name.name.len() - 1].to_string();
+					namespace_filter = vec![];
+					alias = identifier.as_ref().unwrap();
+				} else {
+					// case 2: bring module_name;
+					// case 3: bring module_name as identifier;
+					match module_name.name.as_str() {
+						// If the module name is a built-in module, then we use @winglang/sdk as the library name,
+						// and import the module as a namespace. If the user doesn't specify an identifier, then
+						// we use the module name as the identifier.
+						// For example, `bring fs` will import the `fs` namespace from @winglang/sdk and assign it
+						// to an identifier named `fs`.
+						WINGSDK_CLOUD_MODULE | WINGSDK_FS_MODULE => {
+							library_name = WINGSDK_ASSEMBLY_NAME.to_string();
+							namespace_filter = vec![module_name.name.clone()];
+							alias = identifier.as_ref().unwrap_or(&module_name);
+						}
+						WINGSDK_STD_MODULE => {
+							self.stmt_error(stmt, format!("Redundant import of \"{}\"", WINGSDK_STD_MODULE));
+							return;
+						}
+						_ => {
+							self.stmt_error(stmt, format!("\"{}\" is not a built-in module", module_name.name));
```

Comment:
Eventually this syntax is also going to be used to bring 3rd party `@winglibs` but that's for later...